dbms/src/DataStreams/CheckConstraintsBlockOutputStream.cpp

```diff
@@ -0,0 +1,43 @@
+#include <DataStreams/CheckConstraintsBlockOutputStream.h>
+
+
+namespace DB
+{
+
+void CheckConstraintsBlockOutputStream::write(const Block & block)
+{
+    for (auto & constraint_expr: expressions)
+        if (!checkConstraintOnBlock(block, constraint_expr))
+            throw Exception("Some constraints are not satisfied", ErrorCodes::QUERY_WAS_CANCELLED);
```

Comment:
Better diagnostic is required.

Including:
- what constraint has failed (name, expression);
(to check it, you can evaluate constraints one by one in this case)
- on what row: print row number (indexed from 1 from the beginning of the stream);
- on what row: print all values of related columns.

We need as good error message as possible!